### PR TITLE
MINIFICPP-2214 Fix extension list inconsistencies in build scripts

### PR DESCRIPTION
**Pull request by szaszm**
- Enable MQTT by default in the Windows MSI
- Change all instances of "OS X" and "Mac OS X" to "macOS", to be consistent with Apple's current spelling
- Update the extension list in the README bootstrap example to reflect enabled by default extensions
- Remove abandoned bootstrap.py
- Update bootstrap.sh to treat recently enabled by default extensions as such
- Change win_build_vs.bat CLI switches for enabled by default extensions from an enable switch to an optional disable switch
- Rename win_build_vs.bat internal variables to a more consistent naming scheme
- Change win_build_vs.bat default build platform to x64, with an optional /32 flag to revert to Win32

I didn't do any in-depth refactoring, as it would be wasted effort with the upcoming [MINIFICPP-1797 Python bootstrap rewrite](https://issues.apache.org/jira/browse/MINIFICPP-1797).

--------

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: win_build_vs.bat [hunk 1]
```diff
@@ -65,34 +63,33 @@ for %%x in (%*) do (
...
+    if [%%~x] EQU [/NO_ENCRYPT_CONFIG] set enable_encrypt_config=OFF
+    if [%%~x] EQU [/NO_LUA_SCRIPTING] set enable_lua_scripting=OFF
+    if [%%~x] EQU [/NO_MQTT]          set enable_mqtt=OFF
+    if [%%~x] EQU [/NO_OPC]           set enable_opc=OFF
     if [%%~x] EQU [/OPENWSMAN]        set enable_openwsman=ON
-    if [%%~x] EQU [/OPS]              set enable_ops=ON
+    if [%%~x] EQU [/NO_OPS]           set enable_ops=OFF
     if [%%~x] EQU [/PCAP]             set enable_pcap=ON
-    if [%%~x] EQU [/PYTHON_SCRIPTING] set enable_python_scripting=ON
+    if [%%~x] EQU [/NO_PYTHON_SCRIPTING] set enable_python_scripting=OFF
```
Comment (adamdebreceni) [inline win_build_vs.bat:89]: what crossed my mind is if we could mirror the arguments like `/std:c++20`, as I understand they are still treated as a single token, but they provide something like a key-value argument, having a "NO_" prefix looks like a strong coupling between how we parametrize `win_build_vs.bat` and the default extensions, while `/MQTT:NO` feels different

Comment (szaszm) [reply win_build_vs.bat:89]: I want to keep the changes as minimal as possible. Would `/MQTT` (meaning on) and `/MQTT:OFF` (meaning off) be enough for all features?

Comment (adamdebreceni) [reply win_build_vs.bat:89]: I didn't know it will be replaced soon, in that case it is fine as it is

**Review by adamdebreceni — APPROVED** [review]

File: win_build_vs.bat [hunk 2]
```diff
@@ -65,34 +63,33 @@ for %%x in (%*) do (
     if [%%~x] EQU [/T]                set skiptests=ON
     if [%%~x] EQU [/R]                set skiptestrun=ON
     if [%%~x] EQU [/P]                set cpack=ON
-    if [%%~x] EQU [/K]                set build_kafka=ON
-    if [%%~x] EQU [/J]                set build_JNI=ON
-    if [%%~x] EQU [/S]                set build_SQL=ON
-    if [%%~x] EQU [/C]                set build_coap=ON
-    if [%%~x] EQU [/A]                set build_AWS=ON
-    if [%%~x] EQU [/SFTP]             set build_SFTP=ON
```
Comment (fgerlits) [inline win_build_vs.bat:73]: what happened to `/SFTP`?

File: win_build_vs.bat (hunk 1 above)
Comment (szaszm) [reply win_build_vs.bat:89]: [MINIFICPP-1797 Python bootstrap rewrite](https://issues.apache.org/jira/browse/MINIFICPP-1797)

File: win_build_vs.bat (hunk 2 above)
Comment (szaszm) [reply win_build_vs.bat:73]: Accidentally deleted it, most likely either during the manual conflict resolution with 4e08a94 or somewhere during the automatic rebase process.

Added back in [e859303](https://github.com/apache/nifi-minifi-cpp/pull/1654/commits/e859303f007aff701cccf40b2ef6dab90c1f44c4). I also realized with this change that Windows.md needs to be updated as well, so I did that in [e11810e](https://github.com/apache/nifi-minifi-cpp/pull/1654/commits/e11810edda2c2fd69ac9362c7c344b772b3d640c)

**Review by lordgamez — APPROVED** [review]
LGTM, as I see this also solves https://issues.apache.org/jira/browse/MINIFICPP-2199 so we could include that in the description as well

File: win_build_vs.bat [hunk 3]
```diff
@@ -24,37 +24,35 @@ set scriptdir=%~dp0
...
-set build_kafka=OFF
-set build_coap=OFF
-set build_jni=OFF
-set build_SQL=OFF
-set build_AWS=OFF
-set build_SFTP=OFF
-set build_azure=OFF
+set build_platform=x64
+set enable_kafka=ON
+set enable_coap=OFF
```
[the remark this answers is not in the document]
Comment (szaszm) [reply win_build_vs.bat:29]: Thanks, I didn't notice the duplication here. Removed both the second `set enable_coap=OFF` from here and the second `-DENABLE_COAP=%enable_coap%` from the CMake command line.

**Review by fgerlits — APPROVED** [review]

**Pull request closed without merging**